### PR TITLE
TA#21559 Add module stock_component

### PR DESCRIPTION
**Pull request by ddufresne**
This commit contains only a part of the module.
The remaining code will be added in another PR.

The README will be added after.

### CODE REVIEW

File: stock_component/models/stock_component_line.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
```
Comment (ddufresne) [inline stock_component/models/stock_component_line.py]: Codacy Issue found: [Unused ValidationError imported from odoo.exceptions](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5253555)

File: stock_component/models/stock_location.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
```
Comment (ddufresne) [inline stock_component/models/stock_location.py]: Codacy Issue found: [Unused ValidationError imported from odoo.exceptions](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5253555)

File: stock_component/models/stock_component_line.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
```
Comment (ddufresne) [inline stock_component/models/stock_component_line.py]: Codacy Issue found: [Unused _ imported from odoo](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5253555)

File: stock_component/models/stock_location.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, _
```
Comment (ddufresne) [inline stock_component/models/stock_location.py]: Codacy Issue found: [Unused _ imported from odoo](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5253555)

File: stock_component/models/stock_component_line.py (hunk 3 above)
Comment (foutoucour) [inline stock_component/models/stock_component_line.py]:
```suggestion
from odoo import api, fields, models
```

File: stock_component/models/stock_component_line.py (hunk 1 above)
Comment (foutoucour) [inline stock_component/models/stock_component_line.py]:
```suggestion
```

File: stock_component/models/stock_location.py (hunk 4 above)
Comment (foutoucour) [inline stock_component/models/stock_location.py]:
```suggestion
from odoo import api, fields, models
```

File: stock_component/models/stock_location.py (hunk 2 above)
Comment (foutoucour) [inline stock_component/models/stock_location.py]:
```suggestion
```

File: stock_component/models/stock_production_lot.py [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+        for serial in self:
+            serial.component_ids = serial.mapped("component_line_ids.component_id")
+
+    def _compute_parent_component(self):
+        for serial in self:
+            serial.parent_component_id = serial._get_parent_component()
+
+    def _get_parent_component(self):
+        return self.env["stock.production.lot"].search(
+            [("component_line_ids.component_id", "=", self.id)], limit=1
```
Comment (foutoucour) [inline stock_component/models/stock_production_lot.py:37]: Est ce qu'il peut y en avoir plus d'1 ?

Comment (ddufresne) [reply stock_component/models/stock_production_lot.py:37]: Non, il y a une contrainte qui prévient cela. Je me sens mieux si la limit est là quand même ;)

**Review by foutoucour — APPROVED** [review]

File: stock_component/models/stock_location.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
```
Comment (ddufresne) [inline stock_component/models/stock_location.py]: Codacy Issue found: [Unused fields imported from odoo](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5253555)

File: stock_component/models/stock_component_line.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
```
Comment (ddufresne) [inline stock_component/models/stock_component_line.py]: Codacy Issue found: [Unused api imported from odoo](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5253555)

**Pull request merged**